src/main/java/org/jitsi/jicofo/bridge/BridgeSelector.java

```diff
@@ -210,12 +212,32 @@ synchronized public Bridge selectBridge(
             @NotNull JitsiMeetConference conference,
             String participantRegion)
     {
-        List<Bridge> bridges
-            = getPrioritizedBridgesList().stream()
+        // the list of all known videobridges JIDs ordered by load and *operational* status.
+        ArrayList<Bridge> prioritizedBridges;
+        synchronized (this)
+        {
+            prioritizedBridges = new ArrayList<>(bridges.values());
+        }
+        Collections.sort(prioritizedBridges);
+
+        List<Bridge> candidateBridges
+            = prioritizedBridges.stream()
                 .filter(Bridge::isOperational)
+                .filter(not(Bridge::isInGracefulShutdown))
                 .collect(Collectors.toList());
+
+        // if there's no candidate bridge, we include bridges that are in graceful shutdown mode
+        // (the alternative is to crash the user)
```

Comment:
I think this will work if the strategy selects one of the existing conference bridges. But if it selects a new bridge that is in graceful shutdown, allocating a colibi conference on it will fail. What would you think about being explicit about this (assuming I got this right), and doing something like this?
```
candidateBridges = conference.getBridges().stream().filter(Bridge::isOperational)
```